pkg/networking/cilium/reconciler/reconciler_test.go

```diff
@@ -283,13 +299,19 @@ func TestReconcilerReconcileUpgradeInvalidCiliumInstalledVersion(t *testing.T) {
 		}
 	})
 
+	manifest := buildManifest(tt.WithT, ds, operator)
+	tt.templater.EXPECT().GenerateManifest(tt.ctx, tt.spec).Return(manifest, nil)
+
 	tt.makeCiliumDaemonSetReady()
 	tt.makePreflightDaemonSetReady()
 	tt.makePreflightDeploymentReady()
 
 	result, err := tt.reconciler.Reconcile(tt.ctx, test.NewNullLogger(), tt.client, tt.spec)
 	tt.Expect(result).To(Equal(controller.Result{}))
-	tt.Expect(err).To(MatchError(ContainSubstring("installed cilium DS has an invalid version tag")))
```

Comment:
Ummm I think this is still a valid test case, right?

Instead of changing the expectations, I would change the test data in line 287
`ds.Spec.Template.Spec.Containers[0].Image = "cilium:eksa-invalid-version"`